adserver/views.py

```diff
@@ -413,22 +415,24 @@ def send_to_analytics(self, request, advertisement, message):
     def get(self, request, advertisement_id, nonce):
         """Handles proxying ad views and clicks and collecting metrics on them."""
         advertisement = get_object_or_404(Advertisement, pk=advertisement_id)
-        publisher = advertisement.get_publisher(nonce)
+        offer = Offer.objects.filter(id=nonce).first()
```

Comment:
This can be none. In that case, we won't know the publisher and the view shouldn't count (somebody just made up a nonce/offer). This function as written will 500 since it depends on `offer.publisher`.

Reply:
Yea, I had it as a `get_object_or_404` but it was raising some odd validation errors from Django when passing invalid nonce's. It should return from the `ignore_tracking_reason` when the nonce is invalid, so I will just fix the logic here to let `publisher` be `None`, which should be accounted for.

Reply:
It still throws a 500 if for a URL like `/proxy/view/698/blah/` since `blah` is an invalid UUID. This will cause an issue with all the existing nonces out there when we roll it out (anyone clicking those will trigger a 500). It isn't a huge deal but we can avoid it. I think ideally, it should just not count that for billing and redirect appropriately.

Reply:
Yea, if you have thoughts I'm happy to hear them. Should we be catching that exception and just setting `offer` to `None`?

Reply:
I've added this, and it no longer 500's.